### PR TITLE
fix(MaterialUI): migration of material ui to v6 

### LINKED ISSUE #2453: [FEATURE] <Migrate Material UI and Material React Table>
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Who request this Feature

_No response_

### User Story

Migrate Material UI to v6 for reference follow https://mui.com/material-ui/migration/upgrade-to-v6/
Migrate Material React Table to v3 for supporting material UI v6 version. https://v3.material-react-table.com/docs/getting-started/migrating-to-v3

### Implementation UI

_No response_

### PR DESCRIPTION
**Pull request by kaminderpal**
# Description
Migrate Material UI to v6

Fixes #2453 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

__Add the URL for your deploy!__
https://kaminderpal.github.io/geoview/demos-navigator.html

# Checklist:

- [x] I have build __(rush build)__ and deploy __(rush host)__ my PR
- [x] I have connected the issues(s) to this PR
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have created new issue(s) related to the outcome of this PR is needed
-  ~~I have made corresponding changes to the documentation~~
-  ~~I have added tests that prove my fix is effective or that my feature works~~
-  ~~New and existing unit tests pass locally with my changes~~

- - -
This change is [Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456)

### CODE REVIEW

**Review by jolevesq — CHANGES_REQUESTED** [review]
> Reviewed 12 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456)* status: all files reviewed, 1 unresolved discussion (waiting on @cphelefu and @kaminderpal)

___
*[`packages/geoview-core/src/ui/layout/index.ts` line 1 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456#-O5yTYzYBAhe5IalkWiZ:-O5yTYzZ-nrkzDwi1jl9:b64coh0) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/5121c7b6add21d58b6b55f982bcee209e5165074/packages/geoview-core/src/ui/layout/index.ts#L1)):*
> ```XML
> export { Box, Container, Dialog, DialogActions, DialogTitle, Grid, Link, Stack, DialogContent } from '@mui/material';
> ```

There is no need for grid2 here?

**Review by kaminderpal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456)* status: all files reviewed, 1 unresolved discussion (waiting on @cphelefu and @jolevesq)

___
*[`packages/geoview-core/src/ui/layout/index.ts` line 1 at r1](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456#-O5yTYzYBAhe5IalkWiZ:-O5ycHdi6jhFzEwMtIkt:bafce2m) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/5121c7b6add21d58b6b55f982bcee209e5165074/packages/geoview-core/src/ui/layout/index.ts#L1)):*
<details><summary><i>Previously, jolevesq (Johann Levesque) wrote…</i></summary><blockquote>

There is no need for grid2 here?
</blockquote></details>

we dnt need here, bcz we are already exporting in `grid` dir at same level of layout and exporting in index file.

**Comment by cphelefu** [comment]
Yep. Looks like the UI is broken.

I cant switch tabs on this page.
I am also seeing the exit button.

image

**Comment by cphelefu** [comment]
This change affects all packages. I think we have to spend more time clicking around to make sure nothing is broken. Is there a reason for this version upgrade?

**Review by kaminderpal — COMMENTED** [review]
fixed all above issues, it was due to bcz i miss one file in build and host.

> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456#-:-O61jVzsE2J1MBwjkMdw:bgbpx0i)* status: 11 of 14 files reviewed, 1 unresolved discussion (waiting on @cphelefu and @jolevesq)

File: packages/geoview-core/src/core/components/map-info/map-info.tsx [hunk 1]
```diff
@@ -47,31 +47,30 @@ export function MapInfo(): JSX.Element {
...
         )}
 
-        <Grid container item md={interaction === 'dynamic' ? 11 : 12} spacing={2}>
+        <Grid container spacing={2} size={{ md: interaction === 'dynamic' ? 11 : 12 }}>
           <Grid
             container
-            justifyContent="flex-end"
             sx={{
-              [theme.breakpoints.down('md')]: {
-                marginTop: '-32px',
```
Comment (cphelefu) [inline packages/geoview-core/src/core/components/map-info/map-info.tsx:63]: Why did we remove this code? It accounts for medium screens.

File: packages/geoview-core/src/core/components/map-info/map-info.tsx [hunk 2]
```diff
@@ -47,31 +47,30 @@ export function MapInfo(): JSX.Element {
...
+      <Grid container justifyContent="space-between" size={12}>
         {interaction === 'dynamic' && (
-          <Grid item md={1}>
+          <Grid size={{ md: 1 }}>
             <Attribution />
           </Grid>
         )}
 
-        <Grid container item md={interaction === 'dynamic' ? 11 : 12} spacing={2}>
+        <Grid container spacing={2} size={{ md: interaction === 'dynamic' ? 11 : 12 }}>
```
Comment (cphelefu) [inline packages/geoview-core/src/core/components/map-info/map-info.tsx:57]: @kaminderpal did we validate the removal of spacing here?

**Review by kaminderpal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456)* status: 11 of 17 files reviewed, 3 unresolved discussions (waiting on @cphelefu and @jolevesq)

___
*[`packages/geoview-core/src/core/components/map-info/map-info.tsx` line 57 at r3](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456#gh-1745839377:-O621L7MDTvwQRhKoih1:b2guqqs) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/ed9457732b06a1f05083308e9d58f62af5a29bcf/packages/geoview-core/src/core/components/map-info/map-info.tsx#L57)):*
<details><summary><i>Previously, cphelefu (Christopher Phelefu) wrote…</i></summary><blockquote>

@kaminderpal did we validate the removal of spacing here?
</blockquote></details>

i didnt remove any spacing.

**Review by kaminderpal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456)* status: 11 of 17 files reviewed, 3 unresolved discussions (waiting on @cphelefu and @jolevesq)

___
*[`packages/geoview-core/src/core/components/map-info/map-info.tsx` line 0 at r3](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456#gh-1745837232:-O622jKc9Rb4TugPdU1_:bk403or) ([raw file](https://github.com/Canadian-Geospatial-Platform/geoview/blob/ed9457732b06a1f05083308e9d58f62af5a29bcf/packages/geoview-core/src/core/components/map-info/map-info.tsx#L0)):*
<details><summary><i>Previously, cphelefu (Christopher Phelefu) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/Canadian-Geospatial-Platform/geoview/pull/2456#discussion_r1745837232) to the right spot — sorry!)*

Why did we remove this code? It accounts for medium screens.
</blockquote></details>

changes that i made for grid, fix the issue, and we dnt need -ve margin anymore.

**Review by jolevesq — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Canadian-Geospatial-Platform/geoview/2456)* status: all files reviewed, 2 unresolved discussions (waiting on @cphelefu)

**Pull request merged**